src/crepe/api/AI.h

```diff
@@ -0,0 +1,128 @@
+	};
+
+public:
+	/**
+	 * \param id The id of the game object
+	 * \param max_force The maximum force that can be applied to the entity
+	 */
+	AI(game_object_id_t id, float max_force);
+
+	/**
+	 * \brief Check if a behavior is on (aka activated)
+	 *
+	 * \param behavior The behavior to check
+	 * \return true if the behavior is on, false otherwise
+	 */
+	bool on(BehaviorType behavior) const { return (flags & behavior) == behavior; }
+	//! Turn on the seek behavior
+	void seek_on() { flags |= SEEK; }
+	//! Turn off the seek behavior
+	void seek_off() {
+		if (on(SEEK)) flags ^= SEEK;
+	}
+	//! Turn on the flee behavior
+	void flee_on() { flags |= FLEE; }
+	//! Turn off the flee behavior
+	void flee_off() {
+		if (on(FLEE)) flags ^= FLEE;
+	}
+	//! Turn on the arrive behavior
+	void arrive_on() { flags |= ARRIVE; }
+	//! Turn off the arrive behavior
+	void arrive_off() {
+		if (on(ARRIVE)) flags ^= ARRIVE;
+	}
+	//! Turn on the path follow behavior
+	void path_follow_on() { flags |= PATH_FOLLOW; }
+	//! Turn off the path follow behavior
+	void path_follow_off() {
+		if (on(PATH_FOLLOW)) flags ^= PATH_FOLLOW;
+	}
+
+	/**
+	 * \brief Add a path node (for the path following behavior)
+	 *
+	 * \note The path is not relative to the entity's position (it is an absolute path)
+	 *
+	 * \param node The path node to add
+	 */
+	void add_path_node(vec2 node) { path.push_back(node); }
+	/**
+	 * \brief Make a circle path (for the path following behavior)
+	 *
+	 * \note The path is not relative to the entity's position (it is an absolute path)
+	 *
+	 * \param radius The radius of the circle (in game units)
+	 * \param center The center of the circle (in game units)
+	 * \param start_angle The start angle of the circle (in radians)
+	 * \param clockwise The direction of the circle
+	 */
+	void make_circle_path(float radius, vec2 center = {0, 0}, float start_angle = 0,
+						  bool clockwise = true);
+	/**
+	 * \brief Make an oval path (for the path following behavior)
+	 *
+	 * \note The path is not relative to the entity's position (it is an absolute path)
+	 *
+	 * \param radius_x The x radius of the oval (in game units)
+	 * \param radius_y The y radius of the oval (in game units)
+	 * \param center The center of the oval (in game units)
+	 * \param start_angle The start angle of the oval (in radians)
+	 * \param clockwise The direction of the oval
+	 * \param rotation The rotation of the oval (in radians)
+	 */
+	void make_oval_path(float radius_x, float radius_y, vec2 center = {0, 0},
+						float start_angle = 0, bool clockwise = true, float rotation = 0);
+
+public:
+	//! The maximum force that can be applied to the entity (higher values will make the entity adjust faster)
+	float max_force;
+
+	//! The target to seek or arrive at
+	vec2 seek_target;
+	//! The target to flee from
+	vec2 flee_target;
+	//! The distance at which the entity will start to flee from the target
+	float square_flee_panic_distance = 200.0f * 200.0f;
```

Comment:
waarom 200.0f * 200.0f in plaats van 400.0f?

Reply:
`square_flee_panic_distance` is de afstand vanaf waar een entiteit zal gaan vluchten. Deze afstand wordt genoteerd als `square` omdat dat veel rekenkracht scheelt, deze notatie lijkt mij daarom logischer. De default afstand vanaf waar gevlucht zal worden is dus 200 (game units).
Is dit duidelijk?

Reply:
ah ja dan snap ik het en 200 * 200 is geen 400 :D